src/wallet/wallet.cpp

```diff
@@ -4231,8 +4231,11 @@ util::Result<MigrationResult> MigrateLegacyToDescriptor(const std::string& walle
         // First change to using SQLite
         if (!local_wallet->MigrateToSQLite(error)) return util::Error{error};
 
-        // Do the migration, and cleanup if it fails
-        success = DoMigration(*local_wallet, context, error, res);
+        // Do the migration of keys and scripts for non-blank wallets, and cleanup if it fails
+        success = local_wallet->IsWalletFlagSet(WALLET_FLAG_BLANK_WALLET);
```

Comment:
In 0e7eb54bfe6:

Are we sure we always have unset the blank flag when an address, script or key was imported across all of our previous versions?

Because, if we are not, could change this line for:
```c++
success = !WalletBatch(local_wallet->GetDatabase()).HasLegacyRecords();
```
.
The implementation of `HasLegacyRecords` would be:

```c++
bool WalletBatch::HasLegacyRecords()
{
    return std::any_of(DBKeys::LEGACY_TYPES.begin(), DBKeys::LEGACY_TYPES.end(), [&](const auto& type) { return m_batch->Exists(type); });
}
```

Reply:
> Are we sure we always have unset the blank flag when an address, script or key was imported across all of our previous versions?

I'm pretty sure that it was unset whenever something was imported.

Reply:
In commit "wallet: Skip key and script migration for blank wallets" (8c127ff1edb6b9a607bf1ad247893347252a38e3)

> I'm pretty sure that it was unset whenever something was imported.

This should be true but it would be good to raise an error if the blank flag is set and these records do exist. Seems like it would easy to check for, given the HasLegacyRecords() function. It would document the assumption being made by this code, and provide better error checking.